### PR TITLE
experimental: make it possible to select build rules to use

### PR DESCRIPTION
**Pull request by DavidKorczynski**
It's desirable to isolate the use of a subset of all the builders to speed up development time

### CODE REVIEW

File: experimental/c-cpp/build_generator.py [hunk 1]
```diff
@@ -623,7 +623,23 @@ def match_build_heuristics_on_folder(abspath_of_target: str):
...
+  print("Filtering out build scripts")
+  build_heuristics_to_analyse = os.getenv('BUILD_HEURISTICS', 'all')
+  if build_heuristics_to_analyse != 'all':
+    all_build_heuristics = build_heuristics_to_analyse.split(',')
+    for name in all_build_heuristics:
+      for check in all_checks:
+        if check.name == name:
+          checks_to_test.append(check)
+        else:
+          checks_to_test.append(check)
```
Comment (DonggeLiu) [inline experimental/c-cpp/build_generator.py:637]: Same statement under both `if` and `else`?

File: experimental/c-cpp/build_generator.py [hunk 2]
```diff
@@ -623,7 +623,23 @@ def match_build_heuristics_on_folder(abspath_of_target: str):
       AutogenScannerSH(),
   ]
 
-  for scanner in all_checks:
+  checks_to_test = []
+
+  print("Filtering out build scripts")
+  build_heuristics_to_analyse = os.getenv('BUILD_HEURISTICS', 'all')
+  if build_heuristics_to_analyse != 'all':
```
Comment (DonggeLiu) [inline experimental/c-cpp/build_generator.py:630]: nit: Swap for better readability.
```python
if build_heuristics_to_analyse == 'all':
  checks_to_test = all_checks
else:
  ...
```

**Review by DonggeLiu — APPROVED** [review]

File: experimental/c-cpp/build_generator.py (hunk 1 above)
Comment (DavidKorczynski) [reply experimental/c-cpp/build_generator.py]: Thanks!

File: experimental/c-cpp/build_generator.py (hunk 2 above)
Comment (DavidKorczynski) [reply experimental/c-cpp/build_generator.py]: Thanks! done

File: experimental/c-cpp/build_generator.py (hunk 1 above)
Comment (DavidKorczynski) [reply experimental/c-cpp/build_generator.py]: fixed

**Pull request merged**